### PR TITLE
Update readonly accounts docs

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Docs don't reflect current implementation with regard to readonly/credit-only accounts

#### Summary of Changes
- Move credit-only proposal out of "implemented" folder
- Add implemented doc on readonly accounts

### CODE REVIEW

File: book/src/proposals/credit-only-credit-debit-accounts.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This design covers the handling of credit-only and credit-debit accounts in the
 
 To identify credit-only accounts by program id would require the account to be fetched and loaded from disk. This operation is expensive, and while it is occurring, the runtime would have to reject any transactions referencing the same account.
 
-The proposal introduces a `num_readonly_accounts` field to the transaction structure, and removes the `program_ids` dedicated vector for program accounts.
+The proposal introduces a `num_credit_only_accounts` field to the transaction structure, and removes the `program_ids` dedicated vector for program accounts.
```
Comment (aeyakovenko) [inline book/src/proposals/credit-only-credit-debit-accounts.md]: Why num_credit_only_accouts?  Instead of read only.

Comment (CriesofCarrots) [reply book/src/proposals/credit-only-credit-debit-accounts.md]: Sorry, I should have been more descriptive in the pr comment... I moved the original credit-only design proposal back into "Proposals", as it seemed like there might be impetus to come back around to credit-only again in the future. So I restored the original field names (that got edited as part of my readonly pr). That said, the whole proposal could probably be edited to reflect just the differences between the current readonly implementation and a proposed credit-only implementation.
Or dropped and rewritten at such time as we want to reapproach. Let me know if you have a preference there.

Comment (aeyakovenko) [reply book/src/proposals/credit-only-credit-debit-accounts.md]: The implemented section should reflect what’s implemented.  We can drop the credit only stuff.

Comment (CriesofCarrots) [reply book/src/proposals/credit-only-credit-debit-accounts.md]: Yeah, that’s why I moved it out of implemented.
But sounds good, I’ll delete it altogether.

Comment (aeyakovenko) [reply book/src/proposals/credit-only-credit-debit-accounts.md]: lgtm after deleting the credit only stuff.  I think @garious would prefer a more generalized solution anyways.

**Pull request merged**